### PR TITLE
Adds OpenROAD image generation to the openroad tool driver

### PR DESCRIPTION
**Pull request by gadfort**
Adds:
- image generation at the end of each OpenROAD task: placement, routed nets, heatmaps, clock trees, etc.

### CODE REVIEW

**Review by troibe — APPROVED** [review]
Looks overall good to me.
A couple of questions and one suggestion for the tests.

File: siliconcompiler/tools/openroad/scripts/sc_procs.tcl [hunk 1]
```diff
@@ -148,6 +148,19 @@ proc sc_has_routing {} {
   return false
 }
 
+###########################
+# Check if design has global routing
+###########################
+
+proc sc_has_global_routing {} {
+  foreach net [[ord::get_db_block] getNets] {
+    if { [llength [$net getGuides]] != 0 } {
```
[the remark this answers is not in the document]
Comment (gadfort) [reply siliconcompiler/tools/openroad/scripts/sc_procs.tcl:157]: the global router will generate guides for the net

File: siliconcompiler/tools/openroad/scripts/sc_write_images.tcl [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+  set area []
+  lappend area [ord::dbu_to_microns [expr [$box xMin] - $xmargin]]
+  lappend area [ord::dbu_to_microns [expr [$box yMin] - $ymargin]]
+  lappend area [ord::dbu_to_microns [expr [$box xMax] + $xmargin]]
+  lappend area [ord::dbu_to_microns [expr [$box yMax] + $ymargin]]
+  return $area
+}
+
+proc sc_image_clear_selection {} {
+  for {set i 0} {$i <= 8} {incr i} {
```
[the remark this answers is not in the document]
Comment (gadfort) [reply siliconcompiler/tools/openroad/scripts/sc_write_images.tcl:26]: it's hardcoded into openroad

File: tests/tools/test_openroad.py [hunk 3]
```diff
@@ -45,6 +51,23 @@ def test_openroad(scroot):
...
+    chip = _setup_fifo(scroot)
+    chip.set('tool', 'openroad', 'task', 'floorplan', 'var', 'ord_enable_images', 'true')
+
+    chip.run()
+
+    # check that compilation succeeded
+    assert chip.find_result('def', step='floorplan') is not None
+    assert chip.find_result('odb', step='floorplan') is not None
+
+    assert os.path.exists(os.path.join(chip._getworkdir(step='floorplan', index='0'),
```
[the remark this answers is not in the document]
Comment (gadfort) [reply tests/tools/test_openroad.py:66]: good point

**Pull request merged**